File: Front/src/app/models/claim.ts
export class Claim {
  idClaim!: number;
  decision!: string;
  date!: Date; // Using Date type for date property
  subject!: string;
  status!: string;
  description!: string;
  level!: number;
  title?: ClaimType; // Optional property for claim type
  systemProblem?: SystemProb; // Optional property for system problem
  serviceProblem?: ServiceProb; // Optional property for service problem
}

// Enum definitions
export enum ClaimType {
  SYSTEM = 'SYSTEM',
  SERVICE = 'SERVICE',
  OTHER = 'OTHER',
}

export enum SystemProb {
  SERVER = 'SERVER',
  OPERATIONAL_MALFUNCTION = 'OPERATIONAL_MALFUNCTION',
  UNCORRECT_TIMING = 'UNCORRECT_TIMING',
}

export enum ServiceProb {
  STAFF_SHORTAGE = 'STAFF_SHORTAGE',
  DESORGANISATION = 'DESORGANISATION',
}

File: Back/Claim/src/main/java/com/example/claim/Entity/Claim.java
package com.example.claim.Entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table
@Builder
public class Claim {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idClaim ;
    private String decision;
    private String date;
    private String status;
    private String subject;
    private String description;
    private int level;
    @Enumerated(EnumType.STRING)
    private ClaimType title;

    @Enumerated(EnumType.STRING)
    private SystemProb systemProblem ;

    @Enumerated(EnumType.STRING)
    private ServiceProb serviceProblem ;

    @Lob
    private String attachment;

    @OneToOne(mappedBy = "claim", cascade = CascadeType.ALL)
    private Response response;
}
